pulser/simulation/simulation.py

```diff
@@ -137,14 +138,14 @@ def draw(self) -> None:
     def _extract_samples(self) -> None:
         """Populate samples dictionary with every pulse in the sequence."""
 
-        def prepare_dict() -> Dict[str, np.ndarray]:
+        def prepare_dict() -> dict[str, np.ndarray]:
             # Duration includes retargeting, delays, etc.
             return {'amp': np.zeros(self._tot_duration),
                     'det': np.zeros(self._tot_duration),
                     'phase': np.zeros(self._tot_duration)}
 
         def write_samples(slot: _TimeSlot,
-                          samples_dict: Dict[str, Any]) -> None:
+                          samples_dict: dict[str, np.ndarray]) -> None:
```

Comment:
As for `list` and `Sequence`, you should use `Mapping` instead of `dict` when annotating arguments (import from `collections.abc` too)